chain/standard_cheat_code_contract.go

```diff
@@ -67,13 +67,23 @@ func getStandardCheatCodeContract(tracer *cheatCodeTracer) (*CheatCodeContract,
 
 	// Warp: Sets VM timestamp
 	contract.addMethod(
-		"warp", abi.Arguments{{Type: typeUint64}}, abi.Arguments{},
+		"warp", abi.Arguments{{Type: typeUint256}}, abi.Arguments{},
 		func(tracer *cheatCodeTracer, inputs []any) ([]any, *cheatCodeRawReturnData) {
 			// Maintain our changes until the transaction exits.
-			original := tracer.evm.Context.Time
-			tracer.evm.Context.Time = inputs[0].(uint64)
+			originalTime := tracer.evm.Context.Time
+
+			// Retrieve new timestamp and make sure it is LEQ max value of an uint64
+			newTime := inputs[0].(*big.Int)
+			_, uint64Max := utils.GetIntegerConstraints(false, 64)
```

Comment:
This works, I'm fine with it, but maybe even better would be to make this global, so it doesn't need to be recomputed each time. Should be dirt cheap though, so I'm fine with it.

Reply:
Updated in [8922a67](https://github.com/crytic/medusa/pull/255/commits/8922a6781d101d4486cd688e8dbfe5faa058a21d)